stop_coroutine only accepts Coroutine<()> with a unit return type

Following fails to compile on mquad 0.4.13:

```rust
use macroquad::experimental::coroutines::{start_coroutine,stop_coroutine};
async fn job_unit(){}
async fn job_u32() -> u32{42}
#[macroquad::main("stop_coroutine MWE")]
async fn main() {
    let coroutine_unit = start_coroutine(job_unit());
    stop_coroutine(coroutine_unit);
    let coroutine_u32 = start_coroutine(job_u32());
    stop_coroutine(coroutine_u32);
}
```
Message:

```
error[E0308]: mismatched types
   --> src/main.rs:9:20
    |
9   |     stop_coroutine(coroutine_u32);
    |     -------------- ^^^^^^^^^^^^^ expected `Coroutine`, found `Coroutine<u32>`
    |     |
    |     arguments to this function are incorrect
    |
    = note: expected struct `macroquad::experimental::coroutines::Coroutine<()>`
               found struct `macroquad::experimental::coroutines::Coroutine<u32>`
```

It seems like the reason is the omitted type param in the definition for `stop_coroutine`, but I'm not smart enough to figure out why that ends up setting it to unit instead of failing  altogether.

--- a/src/experimental/coroutines.rs
+++ b/src/experimental/coroutines.rs
@@ -227,7 +227,7 @@ pub fn stop_all_coroutines() {
     context.coroutines.clear();
 }
 
-pub fn stop_coroutine(coroutine: Coroutine) {
+pub fn stop_coroutine<T: 'static + Any>(coroutine: Coroutine<T>) {
     let context = &mut get_context().coroutines_context;
 
     context.coroutines.free(coroutine.id);

